src/System.Private.CoreLib/shared/System/Collections/Generic/List.cs

```diff
@@ -1117,11 +1120,11 @@ private bool MoveNextRare()
                 }
 
                 _index = _list._size + 1;
-                _current = default!;
+                _current = default!; // TODO-NULLABLE: Remove ! when nullable attributes are respected
                 return false;
             }
 
-            public T Current => _current;
+            public T Current => _current!;
```

Comment:
Should this be `MaybeNull`?

Reply:
Correct use of the enumerator will never result in a non-nullable T returning null.  Using Current in other situations (e.g. before the first call to MoveNext) is explicitly called out as having undefined behavior, so I figure it's fine for the annotations to be wrong in that case as well, and in doing so make for less warnings when it's used correctly.